ivy/utils/exceptions.py

```diff
@@ -4,79 +4,48 @@
+from IPython import get_ipython
 
 # Helpers #
 # ------- #
 
 
-def _log_stack_trace_truncated(trace_mode, func_wrapper_trace_mode, buffer):
-    if trace_mode in ["frontend", "ivy"]:
-        buffer.write(
-            "<stack trace is truncated to {} specific files,".format(trace_mode),
-            "call `ivy.set_exception_trace_mode('full')` to view the full trace>",
-        )
-
-    if not func_wrapper_trace_mode:
-        buffer.write(
-            "<func_wrapper.py stack trace is squashed,",
-            "call `ivy.set_show_func_wrapper_trace_mode(True)` in order to view this>",
-        )
-
-
-def _write_new_stack_trace(
-    old_stack_trace, trace_mode, func_wrapper_trace_mode, buffer
-):
-    _log_stack_trace_truncated(trace_mode, func_wrapper_trace_mode, buffer)
-    new_stack_trace = []
-    for st in old_stack_trace:
-        if trace_mode == "full" and not func_wrapper_trace_mode:
-            if "func_wrapper.py" not in repr(st):
-                new_stack_trace.append(st)
-        else:
-            if ivy.trace_mode_dict[trace_mode] in repr(st):
-                if not func_wrapper_trace_mode and "func_wrapper.py" in repr(st):
-                    continue
-                new_stack_trace.append(st)
-    buffer.write("".join(tb.format_list(new_stack_trace)))
-
-
-def _custom_exception_handle(type, value, tb_history):
-    trace_mode = ivy.exception_trace_mode
-    func_wrapper_trace_mode = ivy.show_func_wrapper_trace_mode
-    buffer = io.StringIO()
-    if trace_mode == "none":
-        return
-    if trace_mode == "full" and func_wrapper_trace_mode:
-        print("".join(tb.format_tb(tb_history)))
-    else:
-        _write_new_stack_trace(
-            tb.extract_tb(tb_history), trace_mode, func_wrapper_trace_mode, buffer
-        )
-        print(buffer.getvalue())
-    print(type.__name__ + ":", value)
-
+def configure_stack_trace(traceback):
+    """
+    Configure the stack trace to be displayed in the console.
 
-def _write_traceback_history(buffer):
+    Parameters
+    ----------
+    traceback
+        the traceback object
+    """
+    tb = traceback
     trace_mode = ivy.exception_trace_mode
-    func_wrapper_trace_mode = ivy.show_func_wrapper_trace_mode
-    if trace_mode == "none":
-        return
-    if trace_mode == "full" and func_wrapper_trace_mode:
-        buffer.write("".join(tb.format_tb(sys.exc_info()[2])))
-    else:
-        _write_new_stack_trace(
-            tb.extract_tb(sys.exc_info()[2]),
-            trace_mode,
-            func_wrapper_trace_mode,
-            buffer,
-        )
-    buffer.write(
-        "During the handling of the above exception, another exception occurred:\n"
-    )
-
-
-sys.excepthook = _custom_exception_handle
-
+    show_wrappers = ivy.show_func_wrapper_trace_mode
+    while 1:
+        if not tb.tb_next:
+            break
+        frame = tb.tb_next.tb_frame
+        if trace_mode == "ivy":
+            if "ivy/functional/ivy/" in frame.f_code.co_filename:
```

Comment:
can we instead use `os.path.join('ivy', 'functional', 'ivy')` instead? Different OS have different delimiters (`/`, `\`) and this would be more general.
Also given that we're using the paths in later checks in this function, we can probably also refactor it to compute `os.path.join` once and use it everywhere.